### PR TITLE
Show "Add to Workspace" button in the "The project has been generated" message only when no one of the the workspace folders is predecessor of the project folder.

### PR DESCRIPTION
**Pull request by ira-gordin-sap**
Show "Add to Workspace" button in the "The project has been generated" message only when no one of the the workspace folders is predecessor of the project folder.

### CODE REVIEW

**Comment by slavik-lvovsky** [comment]
What happens when no workspace is opened ?

File: backend/src/vscode-youi-events.ts [hunk 1]
```diff
@@ -65,7 +71,8 @@ export class VSCodeYouiEvents implements YouiEvents {
                 const workspacePath = _.get(vscode, "workspace.workspaceFolders[0].uri.fsPath");
                 // 1. target workspace folder should not already contain target generator folder
                 const foundInWorkspace = _.find(vscode.workspace.workspaceFolders, (wsFolder: vscode.WorkspaceFolder) => {
-                    return targetFolderUri.fsPath === wsFolder.uri.fsPath;
+                    // wsFolder is accessor of targetFolderUri, so targetFolderUri is already shown inside wsFolder
+                    return ((wsFolder.uri.fsPath === targetFolderUri.fsPath) || this.isPredecessorOf(wsFolder.uri.fsPath, targetFolderUri.fsPath));
```
[the remark this answers is not in the document]
Comment (ira-gordin-sap) [reply backend/src/vscode-youi-events.ts:75]: There is some scenario for this check. I don't remember. I'll try to find it.

[the remark this answers is not in the document]
Comment (ira-gordin-sap) [reply backend/src/vscode-youi-events.ts:75]: This is for Windows different disks: C: and D:.

[the remark this answers is not in the document]
Comment (ira-gordin-sap) [reply backend/src/vscode-youi-events.ts:75]: I'll add test for this scenario.

**Comment by ira-gordin-sap** [comment]
> What happens when no workspace is opened ?

I think there is a bug here which was here before my change.
It should show the button "Add to Workspace" I think.
I'll fix it as well.

**Comment by ira-gordin-sap** [comment]
> What happens when no workspace is opened ?

I fixed (removed " && workspacePath" from "if (!foundInWorkspace && workspacePath) {"), but I see this comment there: "// 2. Theia bug: vscode.workspace.workspaceFolders should not be undefined or empty".
Are you sure it is ok to fix it?

**Comment by tomer-epstein** [comment]
> > What happens when no workspace is opened ?
> 
> I think there is a bug here which was here before my change.
> It should show the button "Add to Workspace" I think.
> I'll fix it as well.

There was no bug here before.

File: backend/src/vscode-youi-events.ts [hunk 2]
```diff
@@ -51,6 +52,11 @@ export class VSCodeYouiEvents implements YouiEvents {
         }
     }
 
+    private isPredecessorOf(probablePredecessorPath: string, currentPath: string) {
+        const relativePath = relative(probablePredecessorPath, currentPath);
+        return relativePath && !_.startsWith(relativePath, '..') && !isAbsolute(relativePath);
```
Comment (slavik-lvovsky) [inline backend/src/vscode-youi-events.ts:57]: relative is always not absolute

**Review by slavik-lvovsky — CHANGES_REQUESTED** [review]

**Comment by slavik-lvovsky** [comment]
You need to bump package.json version

**Comment by ira-gordin-sap** [comment]
I  added more tests and explanations as we close with you, @slavik-lvovsky

**Review by slavik-lvovsky — APPROVED** [review]

**Pull request merged**